tests/e2e/unified_test.go

```diff
@@ -171,13 +177,42 @@ func testFixturesInSeries(ctx context.Context, t *testing.T, testPause bool, can
 						}
 					})
 
-					replaceFunc := func(s string) string {
-						result := strings.Replace(s, uniqueID, "uniqueid111111", -1)
-						result = strings.Replace(result, project.ProjectID, "cnrm-user", -1)
+					replaceWellKnownValues := func(s string) string {
+						// Replace project id
+						result := strings.Replace(s, project.ProjectID, "example-project", -1)
+
+						// Replace user info
+						obj := make(map[string]any)
+						if err := json.Unmarshal([]byte(s), &obj); err == nil {
+							toReplace, _, _ := unstructured.NestedString(obj, "user")
+							if len(toReplace) != 0 {
+								result = strings.Replace(result, toReplace, "user@google.com", -1)
+							}
+						}
 						return result
 					}
 
+					unique := make(map[string]bool)
+
 					hook := func(i *cassette.Interaction) error {
+						// Discard failed interactions
+						resCode := i.Response.Code
+						if resCode == 404 || resCode == 400 || resCode == 403 {
+							i.DiscardOnSave = true
+						}
+						// Discard repeated operation retry interactions
+						reqURL := i.Request.URL
+						resBody := i.Response.Body
+
+						if strings.Contains(reqURL, "operations") {
+							sorted, _ := sortJSON(resBody)
```

Comment:
Unless I'm missing something, I think we will not see the "done" operation (?).  But ... if it works, then I guess I'm missing something!

Reply:
To clarify, when we run tests against realGCP, the each CRUD operation has an id and retry transport will try to query the operations every 10s and check the the `done` field(i.e. https://screenshot.googleplex.com/43VLyUJ6QSQHAW4.png) until it becomes `true`(https://screenshot.googleplex.com/BDbaj24DEA68x7T.png).  The retry request/response will be recorded in file if we don't filter them out; these retries won't add any value to our record because they are just repeated logs. If we keep them, with replay mode, we'll need to go through all these retry logs again, which is time-consuming. By removing retry logs, we can speed up our tests with replay mode.